fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneStoredFieldsTest.java

```diff
@@ -0,0 +1,395 @@
+            // query again and compare the other fields
+            queryAndAssertFieldsT(query, "text2", Set.of(Tuple.from(6, 1624L), Tuple.from(7, 1625L)), Set.of("Hello record 2", "Hello record layer 2"));
+            queryAndAssertFieldsT(query, "score", Set.of(Tuple.from(6, 1624L), Tuple.from(7, 1625L)), Set.of(13, 14));
+            queryAndAssertFieldsT(query, "is_seen", Set.of(Tuple.from(6, 1624L), Tuple.from(7, 1625L)), Set.of(false, true));
+            queryAndAssertFieldsT(query, "time", Set.of(Tuple.from(6, 1624L), Tuple.from(7, 1625L)), Set.of(8.123, 9.123));
+
+            if (useOptimizedStoredFieldsFormat) {
+                try (FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, index.getOptions())) {
+                    assertDocCountPerSegment(directory, List.of("_0"), List.of(3));
+                }
+                assertTrue(timer.getCounter(LuceneEvents.Waits.WAIT_LUCENE_GET_STORED_FIELDS).getCount() > 5);
+                assertTrue(timer.getCounter(LuceneEvents.Counts.LUCENE_WRITE_STORED_FIELDS).getCount() >= 3);
+            }
+        }
+    }
+
+    private RecordQuery buildQuery(final String term, final List<String> fields, final String docType) {
+        final QueryComponent filter = new LuceneQueryComponent(term, fields);
+        // Query for full records
+        return RecordQuery.newBuilder()
+                .setRecordType(docType)
+                .setFilter(filter)
+                .build();
+    }
+
+    private void queryAndAssertFields(RecordQuery query, String fieldName, Set<Long> expectedPrimaryKeys, Set<?> expectedFields) throws Exception {
+        queryAndAssertFieldsT(query, fieldName, expectedPrimaryKeys.stream().map(Tuple::from).collect(Collectors.toSet()), expectedFields);
+    }
+
+    private void queryAndAssertFieldsT(RecordQuery query, String fieldName, Set<Tuple> expectedPrimaryKeys, Set<?> expectedFields) throws Exception {
+        RecordQueryPlan plan = planner.plan(query);
+        try (RecordCursor<FDBQueriedRecord<Message>> fdbQueriedRecordRecordCursor = recordStore.executeQuery(plan)) {
+            List<FDBQueriedRecord<Message>> result = fdbQueriedRecordRecordCursor.asList().get();
+            Set<Tuple> actualKeyTuples = result.stream().map(FDBQueriedRecord::getIndexEntry).map(IndexEntry::getPrimaryKey).collect(Collectors.toSet());
+            assertEquals(expectedPrimaryKeys, actualKeyTuples);
+            Set<?> fields = result.stream()
+                    .map(FDBQueriedRecord::getStoredRecord)
+                    .filter(Objects::nonNull) //unnecessary, but helps keep the linter happy
+                    .map(FDBStoredRecord::getRecord)
+                    .map(m -> m.getField(m.getDescriptorForType().findFieldByName(fieldName)))
+                    .collect(Collectors.toSet());
+            assertEquals(expectedFields, fields);
```

Comment:
Since you are converting these to sets, you aren't actually asserting that the field values correspond to the correct documents. i.e. I think all of your tests would pass if the doc ids were randomly mixed up. It would be better to have the method take a `Map<Tuple, ?> expected values` and then assert that the `keySet` is the same as the set of primary keys, and that each one has the right value.

Reply:
Done.